src/EditorFeatures/CSharpTest/UseConditionalExpression/UseConditionalExpressionForReturnTests.cs

```diff
@@ -600,6 +603,34 @@ int M()
             yield return 1;
         }
     }
+}");
+        }
+
+        [WorkItem(27960, "https://github.com/dotnet/roslyn/issues/27960")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseConditionalExpression)]
+        public async Task TestWithNoElseBlockButFollowingYieldReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    void M()
```

Comment:
❗️ Return value should be `IEnumerable<int>`

Reply:
Enh.  :)   i think the behavior is correct, even if the code starts 'incorrect'ish.

Reply:
That's fine, but reviewing the tests I can't tell if the missing diagnostic is due to analysis of `yield return`, or due to the diagnostic not wanting to add a `return` statement within a `void` method. Suppose there was a check that "the code does not change the return type" which fast-path returned for `void` methods.

Reply:
FWIW, this was a non-trivial source of false negatives in StyleCop Analyzers' test suite before we required declaring all compilation errors. Of the couple hundred tests containing compilation errors, more than 10 revealed bugs that failed to be caught by the suite because of an unintended side effect of the compilation error.

Reply:
🔗 https://github.com/dotnet/roslyn/issues/21550

Reply:
Fair enough.  Adding tests.